simulation_ws/src/aws_robomaker_simulation_common/nodes/route_manager

```diff
@@ -48,55 +200,58 @@ class RouteManager():
-        if not poses:
-            rospy.loginfo("Route manager initialized no goals, unable to route")
+        self.map = GoalGenerator()
+        rospy.loginfo("Route manager initialized with random goal generator")
 
-        self.goals = RouteManager.route_modes[self.route_mode](poses)
-        rospy.loginfo("Route manager initialized with %s goals in %s mode", len(poses), self.route_mode)
+        self.bad_goal_counter = 0
 
     def to_move_goal(self, pose):
+        if pose is None:
+            raise ValueError("Goal position cannot be NULL")
+
         goal = MoveBaseGoal()
         goal.target_pose.header.stamp = rospy.Time.now()
         goal.target_pose.header.frame_id = "map"
-        goal.target_pose.pose.position = Point(**pose['pose']['position'])
-        goal.target_pose.pose.orientation = Quaternion(**pose['pose']['orientation'])
+        goal.target_pose.pose.position = pose.position
+        goal.target_pose.pose.orientation = pose.orientation
         return goal
 
+
     def route_forever(self):
         rate = rospy.Rate(1)
         while not rospy.is_shutdown():
             try:
-                rospy.loginfo("Route mode is '%s', getting next goal", self.route_mode)
-                current_goal = self.to_move_goal(next(self.goals))
+                if self.bad_goal_counter > 10:
+                    raise ValueError("Bad goal limit exceeded")
+
+                rospy.loginfo("Quering goal generator for the next goal")
+                current_goal = self.to_move_goal(self.map.get_valid_random_world_pos())
                 rospy.loginfo("Sending target goal: %s", current_goal)
                 self.client.send_goal(current_goal)
-
-                if not self.client.wait_for_result():
-                    rospy.logerr("Move server not ready, will try again...")
-                else:
-                    if self.client.get_result():
-                        rospy.loginfo("Goal done: %s", current_goal)
+                
+                try:
+                    #wait 5sec for global plan to be published. If not, scan for a new goal..
+                    rospy.wait_for_message("/move_base/DWAPlannerROS/global_plan", Path, timeout=5)
+                    
+                    if not self.client.wait_for_result():
+                        rospy.logerr("Move server not ready, will try again...")
+                    else:
+                        if self.client.get_result():
+                            rospy.loginfo("Goal done: %s", current_goal)
+                except rospy.exceptions.ROSException:
+                    self.bad_goal_counter += 1
+                    rospy.logwarn("No plan found for goal. Scanning for a new goal...")
+                
                 rate.sleep()
-            except StopIteration:
-                rospy.loginfo("No goals, stopping route manager")
+            
+            except ValueError:
+                rospy.loginfo("No valid goal was found in the map, stopping route manager")
```

Comment:
Add the error message to the loginfo to make it easier to debug.  For instance, `"No valid goal was found in the map, stopping route manager. Error was: %s"

Reply:
Done